conformance/tests/gatewayclass-observed-generation-bump.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, GatewayClassObservedGenerationBump)
+}
+
+var GatewayClassObservedGenerationBump = suite.ConformanceTest{
```

Comment:
The GatewayClass is part of the environment provided by the person running the conformance tests. I personally don't think we should modify it. Maybe we can ensure that ObservedGeneration in status of the provided GatewayClass matches the generation of the resource though?

Reply:
> Maybe we can ensure that ObservedGeneration in status of the provided GatewayClass matches the generation of the resource though?

We do this already - this test is to ensure the observedGeneration changes when the `spec` changes